lib/broadway_cloud_pub_sub/google_api_client/connection.ex

```diff
@@ -0,0 +1,19 @@
+defmodule BroadwayCloudPubSub.GoogleApiClient.Connection do
```

Comment:
Apologies for the tiny bike shedding but do we need a separate module at all? The implementation for both modules is the same and we don't want to expose this to the user anyway, so I would keep it in the main client module and simply have an option that sets the internal tesla adapter to Tesla.Mock (which is ugly but again, since we don't expose and it is an internal implementation detail, it is totally fine). I would even call the option "__internal_tesla_adapter__". Thoughts?

Reply:
Thanks for the input! That sounds much better.